server/src/modules/post/post.service.ts

```diff
@@ -381,15 +318,7 @@ export class PostService {
 
     // prevent search if query is invalid
     if (tagList.length != tags_unchecked.length) {
-      return [
-        helperFunction.responseHandler(
-          false,
-          422,
-          'Invalid tags used in request',
-          null,
-        ),
-        null,
-      ]
+      throw 'Invalid tags used in request'
```

Comment:
Never throw a literal, always throw Errors instead. I'm trying to figure out why our linting rules don't include this.
See https://eslint.org/docs/rules/no-throw-literal for more info.